pipeline-model-definition/src/test/resources/restart/sameCheckoutMultibranch.groovy

```diff
@@ -0,0 +1,56 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+pipeline {
+    agent any
+
+    stages {
+        stage('skip-on-restart') {
+            steps {
+                echo "This shouldn't show up on second run"
+            }
+        }
+        stage('restart') {
+            steps {
+                script {
+                    if (currentBuild.getNumber() % 2 == 1) {
+                        error("Odd numbered build, failing")
+                    } else {
+                        echo "Even numbered build, success"
+                    }
+                }
+            }
+        }
+        stage('post-restart') {
+            when {
+                expression {
+                    !fileExists("newFile")
```

Comment:
Confused by this. We expect this always to be true, right?

Reply:
Ah, yeah, lurking bit from copy-paste.